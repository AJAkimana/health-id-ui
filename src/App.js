import React from 'react';

const App = () => (
  <div>
    <p>Welcome to Health ID</p>
    <button type="button">Welcome</button>
  </div>
);

export default App;
