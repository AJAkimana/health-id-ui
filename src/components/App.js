import React, { Fragment } from 'react';
import PropTypes from 'prop-types';
import { Switch, Route } from 'react-router-dom';
import AuthContainer from './authentication/Container';
import ResetPassword from './authentication/PasswordReset';
import StepperNav from './setup/Stepper';
import Dashboard from './shared/Dashboard/Dashboard';
import ViewProducts from './stock_control/viewProducts';
import ProductPage from './products/ProductPage';
import UserProfile from './profile/Profile';
import ApproveProductDetail from '../container/products/approveProduct';
import ProductDetail from '../containers/productDetail';
import SellScreenContainer from '../containers/sellScreenContainer';
import AddProduct from './products/AddProduct/AddProduct';
import ImportProduct from './products/ImportProduct/ImportProduct';
import SalesHistory from '../containers/salesHistoryContainer';
import MainSetup from './main_setup/mainSetup';
import MainProfile from './main_setup/mainProfileSetup';
import ManageProfile from './main_setup/manageProfileSetup';
import MainPreferences from './main_setup/mainPreferences';
import MainBusinessInformation from './main_setup/mainBusinessSetup';
import MainBusinessView from './main_setup/mainBusinessSetupView';
import MainOutletSetup from './main_setup/mainOutletSetup';
import MainOutletSetupForm from './main_setup/mainOutletSetupForm';
import MainInvitedUsers from './main_setup/mainInvitedUsers';

const App = ({ session }) => (
  <Fragment>
    <Switch>
      <Route exact path="/" component={AuthContainer} />
      <Route exact path="/register" component={AuthContainer} />
      <Route exact path="/setup" component={StepperNav} />
      <Route exact path="/main_setup" render={() => <MainSetup session={session} />} />
      <Route exact path="/main_setup/profile" render={() => <MainProfile session={session} />} />
      <Route exact path="/main_setup/profile/manage_profile_user" render={() => <ManageProfile session={session} />} />
      <Route exact path="/main_setup/preferences/:outletID" render={() => <MainPreferences session={session} />} />
      <Route exact path="/main_setup/business_information" render={() => <MainBusinessInformation session={session} />} />
      <Route exact path="/main_setup/business_information/:businessID" render={() => <MainBusinessView session={session} />} />
      <Route exact path="/main_setup/outlets_registers" render={() => <MainOutletSetup session={session} />} />
      <Route exact path="/main_setup/outlets_registers/new" render={() => <MainOutletSetupForm session={session} />} />
      <Route exact path="/main_setup/users" render={() => <MainInvitedUsers session={session} />} />
      <Route exact path="/dashboard" render={() => <Dashboard session={session} />} />
      <Route exact path="/reset_password/:uid65/:token" component={ResetPassword} />
      <Route exact path="/products/:id/approve" render={() => <ApproveProductDetail session={session} />} />
      <Route exact path="/products" render={() => <ProductPage session={session} />} />
      <Route exact path="/products/:status" render={() => <ProductPage session={session} />} />
      <Route exact path="/profile" component={UserProfile} />
      <Route exact path="/products/:id/details" render={() => <ProductDetail session={session} />} />
      <Route exact path="/stock" render={() => <ViewProducts session={session} />} />
      <Route exact path="/sell" render={() => <SellScreenContainer session={session} />} />
      <Route exact path="/sell/history" render={() => <SalesHistory session={session} />} />
      <Route exact path="/product/add" render={() => <AddProduct session={session} />} />
      <Route exact path="/product/import" render={() => <ImportProduct session={session} />} />
    </Switch>
  </Fragment>
);

App.propTypes = {
  session: PropTypes.objectOf(PropTypes.object)
};

App.defaultProps = {
  session: {}
};

export default App;
