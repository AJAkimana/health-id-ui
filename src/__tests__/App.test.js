import React from 'react';
import App from '../App';
import SetUp from '../utils/TestSetup';

const { shallow } = SetUp();

describe('initial react test', () => {
  it('should render without crashing', () => {
    const component = shallow(<App />);
    expect(component).toMatchSnapshot();
  });
});
