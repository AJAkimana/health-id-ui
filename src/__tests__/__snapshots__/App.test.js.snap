// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`initial react test should render without crashing 1`] = `ShallowWrapper {}`;
